### PR TITLE
YD-213 Prevent information leakage through error codes

### PR DESCRIPTION
**Pull request by Bert-R**
It shouldn't be possible to see whether a mobile number exists, except during the short moment the new device request exists.

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy [hunk 1]
```diff
@@ -61,13 +72,17 @@ class AddDeviceTest extends AbstractAppServiceIntegrationTest
...
 
 		then:
-		response.status == 400
+		responseWrongPassword.status == 400
+		responseWrongPassword.responseData.code == "error.decrypting.data"
 		def getResponseAfter = appService.getNewDeviceRequest(richard.mobileNumber, userSecret)
 		getResponseAfter.status == 200
 		getResponseAfter.responseData.userPassword == richard.password
+		responseWrongMobileNumber.status == 400
+		responseWrongMobileNumber.responseData.code == "error.decrypting.data"
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy]: Would it not be more logical to have "error.wrong.mobile.number.or.password" in both cases?

File: appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy [hunk 2]
```diff
@@ -61,13 +72,17 @@ class AddDeviceTest extends AbstractAppServiceIntegrationTest
...
 
 		when:
-		def response = appService.setNewDeviceRequest(richard.mobileNumber, "foo", "Some secret")
+		def responseWrongPassword = appService.setNewDeviceRequest(richard.mobileNumber, "foo", "Some secret")
+		def responseWrongMobileNumber = appService.setNewDeviceRequest("+31690609181", "foo", "Some secret")
 
 		then:
-		response.status == 400
+		responseWrongPassword.status == 400
+		responseWrongPassword.responseData.code == "error.decrypting.data"
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy]: Would it not be more logical to have "error.wrong.mobile.number.or.password" in both cases?

File: appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy (hunk 1 above)
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy]: We're simply returning the decryption exception to the client. We could catch that and throw a different exception, but I wonder whether that's worth the effort.

Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AddDeviceTest.groovy]: Well, the code now is a bit strange with a passed exception supplier. It becomes easier with just catching an exception.

**Pull request merged**